src/awkward/types/recordtype.py

```diff
@@ -212,16 +211,11 @@ def _is_equal_to(self, other: Any, all_parameters: bool) -> bool:
             if set(self._fields) != set(other._fields):
                 return False
 
-            self_contents = [self.content(f) for f in self._fields]
-            other_contents = [other.content(f) for f in other._fields]
-
-            return any(
-                all(
-                    this._is_equal_to(that, all_parameters)
-                    for this, that in zip(self_contents, contents)
-                )
-                for contents in permutations(other_contents)
+            return all(
+                content._is_equal_to(other.content(field), all_parameters)
+                for field, content in zip(self._fields, self._contents)
             )
```

Comment:
This is wrong!

Reply:
The original logic permuted the contents, but as we have named fields (or positions), we should not do this. Otherwise, `{x: int64, y: complex128}` and `{x: complex128, y: int64}` would be considered equal!

I brought in permutations after working on `enforce_type` and my mind must have been full of rubbish!

Reply:
Yes,

1. `{x: int64, y: complex128} == {x: complex128, y: int64}` is bad and
2. `{x: int64, y: complex128} == {y: complex128, x: int64}` is good.

Although it doesn't look to me like that's what the code above is doing. It's getting a correlated `field, content` pair from `self` and comparing this `content` with `other.content(field)`, which ought to be case (2) above. As long as some code before this also ensures that `set(self.fields) == set(other.fields)`, that is.